src/Image_Backend.php

```diff
@@ -84,87 +78,61 @@ public function setImageResource($resource);
      * @return array Tuple associative array (Filename, Hash, Variant) Unless storing a variant, the hash
      * will be calculated from the given data.
      */
-    public function writeToStore(AssetStore $assetStore, $filename, $hash = null, $variant = null, $config = []);
+    public function writeToStore(AssetStore $assetStore, string $filename, ?string $hash = null, ?string $variant = null, array $config = []): array;
 
     /**
      * Write the backend to a local path
      *
      * @param string $path
      * @return bool if the write was successful
      */
-    public function writeTo($path);
+    public function writeTo(string $path): bool;
 
     /**
      * Set the quality to a value between 0 and 100
-     *
-     * @param int $quality
      */
-    public function setQuality($quality);
+    public function setQuality(int $quality): static;
 
     /**
      * Resize an image, skewing it as necessary.
-     *
-     * @param int $width
-     * @param int $height
-     * @return static
      */
-    public function resize($width, $height);
+    public function resize(int $width, int $height): static;
 
     /**
-     * Resize the image by preserving aspect ratio. By default, it will keep the image inside the maxWidth
-     * and maxHeight. Passing useAsMinimum will make the smaller dimension equal to the maximum corresponding dimension
-     *
-     * @param int $width
-     * @param int $height
-     * @param bool $useAsMinimum If true, image will be sized outside of these dimensions.
-     * If false (default) image will be sized inside these dimensions.
-     * @return static
+     * Resize the image by preserving aspect ratio. By default, the image cannot be resized to be larger
+     * than its current size.
+     * Passing true to useAsMinimum will allow the image to be scaled up.
```

Comment:
It took me a while to wrap my head around this - reworded so other devs don't have to go through the same trials.